### PR TITLE
ISDB-405

### PR DESCRIPTION
**Pull request by parvezhmcts**
- Added user role for cafcass

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/ISDB-405

### Change description ###

user role for cafcass

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x No
```

### CODE REVIEW

File: definitions/private-law/json/AuthorisationCaseState.json [hunk 1]
```diff
@@ -397,5 +397,26 @@
     "CaseStateID": "GATE_KEEPING",
     "UserRole": "caseworker-caa",
     "CRUD": "CRU"
+  },
```
Comment (AlokDatta) [inline definitions/private-law/json/AuthorisationCaseState.json:400]: We need to add access to caseType as well via authorisationCaseType.json

File: definitions/private-law/json/AuthorisationCaseField/PartiesTab/AuthorisationCaseField.json [hunk 2]
```diff
@@ -34,6 +34,13 @@
     "UserRole": "caseworker-privatelaw-superuser",
     "CRUD": "R"
   },
+  {
+    "LiveFrom": "29/09/2022",
+    "CaseTypeID": "PRLAPPS",
+    "CaseFieldID": "partiesTabLabel",
```
Comment (AlokDatta) [inline definitions/private-law/json/AuthorisationCaseField/PartiesTab/AuthorisationCaseField.json:40]: Labels wont be stored

File: definitions/private-law/json/AuthorisationCaseField/SendAndReplyToMessages/AuthorisationCaseField.json [hunk 3]
```diff
@@ -6,47 +6,96 @@
     "UserRole": "caseworker-privatelaw-courtadmin",
     "CRUD": "CRU"
   },
+  {
+    "LiveFrom": "29/09/2022",
+    "CaseTypeID": "PRLAPPS",
+    "CaseFieldID": "chooseSendOrReply",
```
Comment (AlokDatta) [inline definitions/private-law/json/AuthorisationCaseField/SendAndReplyToMessages/AuthorisationCaseField.json:12]: Dont think you need this?

File: definitions/private-law/json/AuthorisationCaseField/ConfidentialTab/AuthorisationCaseField.json [hunk 4]
```diff
@@ -62,11 +69,25 @@
...
     "LiveFrom": "02/12/2021",
     "CaseTypeID": "PRLAPPS",
     "CaseFieldID": "childrenConfidentialDetails",
     "UserRole": "caseworker-privatelaw-la",
     "CRUD": "CRUD"
+  },
+  {
+    "LiveFrom": "29/09/2022",
+    "CaseTypeID": "PRLAPPS",
+    "CaseFieldID": "childrenConfidentialDetails",
```
[the remark this answers is not in the document]
Comment (parvezhmcts) [reply definitions/private-law/json/AuthorisationCaseField/ConfidentialTab/AuthorisationCaseField.json:89]: revert changes.

File: definitions/private-law/json/AuthorisationCaseField/ConfidentialTab/AuthorisationCaseField.json [hunk 5]
```diff
@@ -41,6 +41,13 @@
     "UserRole": "caseworker-privatelaw-courtadmin",
     "CRUD": "CRUD"
   },
+  {
+    "LiveFrom": "29/09/2022",
+    "CaseTypeID": "PRLAPPS",
+    "CaseFieldID": "fl401ChildrenConfidentialDetails",
+    "UserRole": "caseworker-privatelaw-cafcass",
```
[the remark this answers is not in the document]
Comment (parvezhmcts) [reply definitions/private-law/json/AuthorisationCaseField/ConfidentialTab/AuthorisationCaseField.json:48]: revert changes.

File: definitions/private-law/json/AuthorisationCaseField/CourtNav/AuthorisationCaseField.json [hunk 6]
```diff
@@ -48,6 +48,13 @@
     "UserRole": "courtnav",
     "CRUD": "CRU"
   },
+  {
+    "LiveFrom": "29/09/2022",
+    "CaseTypeID": "PRLAPPS",
+    "CaseFieldID": "isCourtNavCase",
+    "UserRole": "caseworker-privatelaw-cafcass",
```
[the remark this answers is not in the document]
Comment (parvezhmcts) [reply definitions/private-law/json/AuthorisationCaseField/CourtNav/AuthorisationCaseField.json:55]: reverted.

File: definitions/private-law/json/AuthorisationCaseField/CourtNav/AuthorisationCaseField.json [hunk 7]
```diff
@@ -89,5 +96,12 @@
     "CaseFieldID": "courtNavUploadedDocs",
     "UserRole": "courtnav",
     "CRUD": "CRU"
+  },
+  {
+    "LiveFrom": "29/09/2022",
+    "CaseTypeID": "PRLAPPS",
+    "CaseFieldID": "courtNavUploadedDocs",
```
[the remark this answers is not in the document]
Comment (parvezhmcts) [reply definitions/private-law/json/AuthorisationCaseField/CourtNav/AuthorisationCaseField.json:103]: reverted.

File: definitions/private-law/json/AuthorisationCaseField/ManageOrders/AuthorisationCaseField.json [hunk 8]
```diff
@@ -146,6 +174,13 @@
     "UserRole": "caseworker-privatelaw-systemupdate",
     "CRUD": "CRU"
   },
+  {
+    "LiveFrom": "31/03/2022",
+    "CaseTypeID": "PRLAPPS",
+    "CaseFieldID": "wasTheOrderApprovedAtHearing",
+    "UserRole": "caseworker-privatelaw-cafcass",
```
[the remark this answers is not in the document]
Comment (parvezhmcts) [reply definitions/private-law/json/AuthorisationCaseField/ManageOrders/AuthorisationCaseField.json:181]: deleted above as suggested.

File: definitions/private-law/json/AuthorisationCaseField/MessagesTab/AuthorisationCaseField.json [hunk 9]
```diff
@@ -34,6 +34,13 @@
     "UserRole": "caseworker-privatelaw-judge",
     "CRUD": "R"
   },
+  {
+    "LiveFrom": "29/09/2022",
+    "CaseTypeID": "PRLAPPS",
+    "CaseFieldID": "messagesLabel",
```
[the remark this answers is not in the document]
Comment (parvezhmcts) [reply definitions/private-law/json/AuthorisationCaseField/MessagesTab/AuthorisationCaseField.json:40]: deleted..

File: definitions/private-law/json/AuthorisationCaseField/PartiesTab/AuthorisationCaseField.json (hunk 2 above)
Comment (parvezhmcts) [reply definitions/private-law/json/AuthorisationCaseField/PartiesTab/AuthorisationCaseField.json:40]: deleted.

File: definitions/private-law/json/AuthorisationCaseField/SendAndReplyToMessages/AuthorisationCaseField.json (hunk 3 above)
Comment (parvezhmcts) [reply definitions/private-law/json/AuthorisationCaseField/SendAndReplyToMessages/AuthorisationCaseField.json:12]: reverted.

File: definitions/private-law/json/AuthorisationCaseState.json (hunk 1 above)
Comment (parvezhmcts) [reply definitions/private-law/json/AuthorisationCaseState.json:400]: done

File: definitions/private-law/json/AuthorisationCaseState.json [hunk 10]
```diff
@@ -397,5 +397,26 @@
...
+    "UserRole": "caseworker-privatelaw-cafcass",
+    "CRUD": "RU"
+  },
+  {
+    "LiveFrom": "29/09/2022",
+    "CaseTypeID": "PRLAPPS",
+    "CaseStateID": "CASE_ISSUE",
+    "UserRole": "caseworker-privatelaw-cafcass",
+    "CRUD": "RU"
+  },
```
[the remark this answers is not in the document]
Comment (parvezhmcts) [reply definitions/private-law/json/AuthorisationCaseState.json:414]: ok

File: definitions/private-law/json/AuthorisationCaseEvent/ApplicantsDetails/AuthorisationCaseEvent.json [hunk 11]
```diff
@@ -40,5 +40,12 @@
     "CaseEventID": "applicantsDetails",
     "UserRole": "caseworker-privatelaw-superuser",
     "CRUD": "R"
+  },
+  {
+    "LiveFrom": "29/09/2022",
+    "CaseTypeID": "PRLAPPS",
+    "CaseEventID": "applicantsDetails",
+    "UserRole": "caseworker-privatelaw-cafcass",
+    "CRUD": "RU"
```
[the remark this answers is not in the document]
Comment (parvezhmcts) [reply definitions/private-law/json/AuthorisationCaseEvent/ApplicantsDetails/AuthorisationCaseEvent.json:49]: reverted.

File: definitions/private-law/json/AuthorisationCaseEvent/Respondent/AuthorisationCaseEvent.json [hunk 12]
```diff
@@ -48,6 +48,13 @@
     "UserRole": "caseworker-privatelaw-systemupdate",
     "CRUD": "CRUD"
   },
+  {
+    "LiveFrom": "29/09/2022",
+    "CaseTypeID": "PRLAPPS",
+    "CaseEventID": "legalRepresentation",
+    "UserRole": "caseworker-privatelaw-cafcass",
+    "CRUD": "RU"
+  },
```
[the remark this answers is not in the document]
Comment (parvezhmcts) [reply definitions/private-law/json/AuthorisationCaseEvent/Respondent/AuthorisationCaseEvent.json:57]: reverted.

File: definitions/private-law/json/AuthorisationCaseField/CaseInvite/AuthorisationCaseField.json [hunk 13]
```diff
@@ -26,5 +26,12 @@
     "CaseFieldID": "respondentCaseInvites",
     "UserRole": "caseworker-privatelaw-systemupdate",
     "CRUD": "CRUD"
+  },
+  {
+    "LiveFrom": "29/09/2022",
+    "CaseTypeID": "PRLAPPS",
+    "CaseFieldID": "respondentCaseInvites",
+    "UserRole": "caseworker-privatelaw-cafcass",
+    "CRUD": "RU"
```
[the remark this answers is not in the document]
Comment (parvezhmcts) [reply definitions/private-law/json/AuthorisationCaseField/CaseInvite/AuthorisationCaseField.json:35]: reverted.

**Review by hmcts-kodatala — APPROVED** [review]
OK

**Pull request merged**